cli/mobycli/exec.go

```diff
@@ -131,6 +122,25 @@ func RunDocker(childExit chan bool, args ...string) error {
 	return cmd.Run()
 }
 
+func comDockerCli() string {
+	if v := os.Getenv("DOCKER_COM_DOCKER_CLI"); v != "" {
+		return v
+	}
+
+	execBinary := findBinary(ComDockerCli)
```

Comment:
Nasty detail on Windows. It seems that `findBinary()` needs to append the `.exe` extension to successfully find the binary in the directory of the `docker.exe` binary. As it only checked `com.docker.cli` without extension it failed and then the `LookPath()` is called, and then the `com.docker.cli.exe` could be found in any folder of my current PATH.

<img width="1348" alt="Screenshot 2022-07-27 at 12 38 03" src="https://user-images.githubusercontent.com/207759/181227439-9631d613-4c9c-4dca-9db7-6a3e3394ce62.png">

Reply:
This is only needed for the Windows binary `docker.exe`, the Linux binary that we use in the WSL integration already works fine, I successfully tested it in a WSL distro on my Windows machine.

Reply:
Good catch, thanks. I am still confused by which Go APIs work without having to append .exe and which don't. Apparently the exec-related API's and LookPath don't...

Reply:
Oh, I see, it's the os.Stat call that needs it. At any rate the current fix is probably best to make sure we only run `com.docker.cli.exe` and not a .bat, etc...